### PR TITLE
chore: Switch to the MPL 2.0 license

### PR DESCRIPTION
**Pull request by binarylogic**
This switches Vector to the [Mozilla Public License 2.0](https://www.mozilla.org/en-US/MPL/2.0/). This is a weak copyleft license that follows other popular open-source projects, such as [Terraform](https://github.com/hashicorp/terraform/blob/master/LICENSE) and [Consul](https://github.com/hashicorp/consul/blob/master/LICENSE). The [MPL 2.0 FAQ](https://www.mozilla.org/en-US/MPL/2.0/FAQ/) describes the license well:

> Like all other free and open source software, software available under the MPL
> is available for anyone (including individuals and companies) to use for any
> purpose. The MPL only creates obligations for you if you want to distribute
> the software outside your organization.

An announcement/highlight will be included with Vector's `0.11.0` release.

I will leave this PR open for further discussion. Anyone is welcome to chime in.

### CODE REVIEW

**Comment by binarylogic** [comment]
I've updated the language in `CONTRIBUTING.md` to reflect the new license and be less specific, delegating that responsibility to the license itself.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -793,16 +793,8 @@ Trivial changes, such as spelling fixes, do not need to be signed.
...
-may be considered an alternate CLA.
-
-The existence of section 5 of the Apache License is proof that the Apache
-License is intended to be usable without CLAs. Users need for the code to be
-open-source, with all the legal rights that imply, but it is the open source
-license that provides this. The Apache License provides very generous
-copyright permissions from contributors, and contributors explicitly grant
-patent licenses as well. These rights are granted to everyone.
+project is the license under which the contribution is made. However, the DCO
+in conjunction with the Vector License may be considered an alternate CLA.
```
Comment (bruceg) [inline CONTRIBUTING.md:797]: This is not really part of this PR, but I looked at the full text of the CONTRIBUTING file, and the acronym CLA is never spelled out.

Comment (binarylogic) [reply CONTRIBUTING.md:797]: That's because CLA is a TLA. I'll update that.

**Comment by BrittanyIstenes** [comment]
Have you let all of your past contributors know that you have made this license change?

**Comment by binarylogic** [comment]
Hi Brittany, we have an announcement teed up for our next release. You can see the highlight [here](https://vector.dev/highlights/2020-08-31-mpl-2-0-license). Are there any specific concerns you have with the license? I want to make sure we are not misunderstanding something that would prohibit usage. To our knowledge, the MPL 2.0 license does not restrict usage or distribution in any way, it only requires that direct Vector code modification (not code around it) be shared back with the project. Happy to answer any questions or discuss further.

**Comment by ignoramous** [comment]
> Have you let all of your past contributors know that you have made this license change?

I don't think that is required since MPLv2 is a stronger license in terms of copyleft than Apache v2 whilst retaining all its properties.

If this project was moving from MPLv2 to Apache v2 or from GPLv3 to MPLv2 that'd have required approval from all contributors (provided they retain copyright over the code they PR in the first place, that is, aren't signees of rather restrictive [CLAs](https://en.wikipedia.org/wiki/Contributor_License_Agreement)).
